cpp/include/cugraph/edge_properties.hpp

```diff
@@ -0,0 +1,210 @@
+    return func(prop);
+  } else if (std::holds_alternative<raft::device_span<double const>>(property)) {
+    auto& prop = std::get<raft::device_span<double const>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<raft::device_span<int32_t const>>(property)) {
+    auto& prop = std::get<raft::device_span<int32_t const>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<raft::device_span<int64_t const>>(property)) {
+    auto& prop = std::get<raft::device_span<int64_t const>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<raft::device_span<size_t const>>(property)) {
+    auto& prop = std::get<raft::device_span<size_t const>>(property);
+    return func(prop);
+  }
+
+  CUGRAPH_FAIL("unsupported variant type -- shouldn't happen");
+}
+
+template <typename edge_t, typename func_t>
+auto variant_type_dispatch(edge_property_views_t<edge_t>& property, func_t func)
+{
+  if (std::holds_alternative<cugraph::edge_property_view_t<edge_t, float const*>>(property)) {
+    auto& prop = std::get<cugraph::edge_property_view_t<edge_t, float const*>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<cugraph::edge_property_view_t<edge_t, double const*>>(
+               property)) {
+    auto& prop = std::get<cugraph::edge_property_view_t<edge_t, double const*>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<cugraph::edge_property_view_t<edge_t, int32_t const*>>(
+               property)) {
+    auto& prop = std::get<cugraph::edge_property_view_t<edge_t, int32_t const*>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<cugraph::edge_property_view_t<edge_t, int64_t const*>>(
+               property)) {
+    auto& prop = std::get<cugraph::edge_property_view_t<edge_t, int64_t const*>>(property);
+    return func(prop);
+  } else if (std::holds_alternative<cugraph::edge_property_view_t<edge_t, size_t const*>>(
+               property)) {
+    auto& prop = std::get<cugraph::edge_property_view_t<edge_t, size_t const*>>(property);
+    return func(prop);
+  }
+
+  CUGRAPH_FAIL("unsupported variant type -- shouldn't happen");
+}
+
+struct variant_size {
+  template <typename T>
+  size_t operator()(rmm::device_uvector<T> const&) const
+  {
+    return sizeof(T);
+  }
+  template <typename T>
+  size_t operator()(raft::device_span<T> const&) const
+  {
+    return sizeof(T);
+  }
+  template <typename T>
+  size_t operator()(raft::device_span<T const> const&) const
+  {
+    return sizeof(T);
+  }
+};
+
+inline numeric_device_span_t make_numeric_device_span(numeric_device_uvector_t& v)
+{
+  return variant_type_dispatch(v, [](auto& v) {
+    using T = typename std::remove_reference<decltype(v)>::type::value_type;
+    return static_cast<numeric_device_span_t>(raft::device_span<T>(v.data(), v.size()));
+  });
+}
+
+inline std::vector<numeric_device_span_t> make_numeric_device_span(
```

Comment:
This makes vector of device span objects, I think this name is misleading.

Reply:
Appended `_vector` in next push.